### PR TITLE
update m2 integration

### PR DESCRIPTION
**Pull request by wuqiten**
Merge m2 update from AKS-Edge AIO quick start M2 script
https://github.com/Azure/AKS-Edge/commit/7f4a4818f5e4e5fad0b3279aa976db1c3dd1da53#diff-8b1f1c59d3185cf64af0038defa5e67b1af745a032ef0af39152397500800845

### CODE REVIEW

File: artifacts/LogonScript.ps1 [hunk 1]
```diff
@@ -246,6 +365,9 @@ $aideuserConfig = @"
         "ServicePrincipalName": "aksedge-sp",
         "Location": "$Location",
         "CustomLocationOID":"$CustomLocationOid",
+        "EnableWorkloadIdentity": true,
```
Comment (vascoveigams) [inline artifacts/LogonScript.ps1:368]: Does this work with the Microsoft tenant?
Can you create clusters and validate e2e pass and so on?
From a Monday meeting, the support for Microsoft tenant is being worked on, and expected within a week or so

Comment (wuqiten) [reply artifacts/LogonScript.ps1:368]: Is Microsoft tenant same with Corp tenant where the subscription Common Infra, Int Release... at?

Comment (vascoveigams) [reply artifacts/LogonScript.ps1:368]: Yes. The other tenant would be AzureStack
You can find the tenant if you do account show or go to entra

Comment (wuqiten) [reply artifacts/LogonScript.ps1:368]: i tested in Microsoft tenant with subscription bd7b1ca2-036a-4ca9-990e-79837f39707e https://dev.azure.com/msazure/One/_build/results?buildId=105325085&view=results
and testing in BareMetal now

**Review by vascoveigams — APPROVED** [review]

**Pull request merged**